src/m365/file/commands/file-move.ts

```diff
@@ -0,0 +1,240 @@
+        newName: typeof args.options.newName !== 'undefined',
+        nameConflictBehavior: typeof args.options.nameConflictBehavior !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '-u, --webUrl <webUrl>' },
+      { option: '-s, --sourceUrl <sourceUrl>' },
+      { option: '-t, --targetUrl <targetUrl>' },
+      { option: '--newName [newName]' },
+      { option: '--nameConflictBehavior [nameConflictBehavior]', autocomplete: this.nameConflictBehaviorOptions }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.nameConflictBehavior && this.nameConflictBehaviorOptions.indexOf(args.options.nameConflictBehavior) === -1) {
+          return `${args.options.nameConflictBehavior} is not a valid nameConflictBehavior value. Allowed values: ${this.nameConflictBehaviorOptions.join(', ')}.`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const { webUrl, sourceUrl, targetUrl, nameConflictBehavior, newName, verbose } = args.options;
+      const sourcePath: string = this.getAbsoluteUrl(webUrl, sourceUrl);
+      const destinationPath: string = this.getAbsoluteUrl(webUrl, targetUrl);
+
+      if (verbose) {
+        logger.logToStderr(`Moving file '${sourcePath}' to '${destinationPath}'...`);
+      }
+
+      const { siteId, driveId, itemId } = await this.getDriveIdAndItemId(webUrl, sourcePath, sourceUrl, logger, verbose);
+      const { targetDriveId, targetItemId } = await this.getTargetDriveAndItemId(webUrl, targetUrl, logger, verbose);
+      const requestOptions: CliRequestOptions = this.getRequestOptions(targetDriveId, targetItemId, newName, sourcePath);
+
+      const isSameDrive: boolean = driveId === targetDriveId;
+      const apiUrl =
+        isSameDrive
+          ? `${this.resource}/v1.0/sites/${siteId}/drives/${driveId}/items/${itemId}`
+          : `${this.resource}/v1.0/sites/${siteId}/drives/${driveId}/items/${itemId}/copy`;
+
+      const queryParameters: string = nameConflictBehavior && nameConflictBehavior !== 'fail'
+        ? `@microsoft.graph.conflictBehavior=${nameConflictBehavior}`
+        : '';
+      const urlWithQuery = `${apiUrl}${queryParameters ? `?${queryParameters}` : ''}`;
+
+      await this.sendRequest(urlWithQuery, requestOptions, isSameDrive, logger);
+
+      if (!isSameDrive) {
+        const itemUrl = `${this.resource}/v1.0/sites/${siteId}/drives/${driveId}/items/${itemId}`;
+        await request.delete({ url: itemUrl, headers: requestOptions.headers });
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async sendRequest(url: string, requestOptions: CliRequestOptions, isPatch: boolean, logger: Logger): Promise<any> {
```

Comment:
From a readability perspective: Let's remove this private function and just move the contents to the commandAction function. It's actually just 2 or 3 lines of code.